### ISSUE TITLE
Property overriding does not work

### ISSUE THREAD

**Issue by sbordet** [eclipse/jetty.project#6017]
**Jetty version**
10.0.x

**Description**
These commands:

```
$ java -jar $JETTY_HOME/start.jar --add-modules=ssl,https,test-keystore
$ java -jar $JETTY_HOME/start.jar jetty.sslContext.keyStorePath=bogus
```

produce:

```
java.lang.IllegalStateException: /tmp/jetty.base/etc/test-keystore.p12 is not a valid keystore
```

The wrong property value is reported in the error message.

But furthermore:

```
$ java -jar $JETTY_HOME/start.jar jetty.sslContext.keyStorePassword=bogus
```

starts Jetty without errors.

There is something wrong in how properties are handled.

**Comment by sbordet** [comment]
Works properly in Jetty 9.4.x.

**Comment by sbordet** [comment]
@joakime any idea?

**Comment by joakime** [comment]
Isn't this related to #5915 ?

**Comment by sbordet** [comment]
@joakime maybe -- the `test-keystore` module does have an `[ini]` section that seems to affect this issue and also #5990.

**Comment by joakime** [comment]
Could be, the `--list-config` shows ...

```
Enabled Modules:
----------------
    0) resources       transitive provider of resources for logging-jetty
    1) logging/slf4j   transitive provider of logging/slf4j for logging-jetty
                       dynamic dependency of logging-jetty
    2) logging-jetty   transitive provider of logging for threadpool
                       transitive provider of logging for bytebufferpool
                       transitive provider of logging for server
    3) bytebufferpool  transitive provider of bytebufferpool for server
                       init template available with --add-module=bytebufferpool
    4) threadpool      transitive provider of threadpool for server
                       init template available with --add-module=threadpool
    5) server          transitive provider of server for ssl
                       init template available with --add-module=server
    6) ssl             ${jetty.base}/start.d/ssl.ini
    7) https           ${jetty.base}/start.d/https.ini
    8) test-keystore   ${jetty.base}/start.d/test-keystore.ini

...(snip)...

Properties:
-----------
 bouncycastle.version = 1.62
 java.version = 11.0.10
 java.version.major = 11
 java.version.micro = 10
 java.version.minor = 0
 java.version.platform = 11
 jetty.base = /home/joakim/code/jetty/distros/bases/prop-base
 jetty.base.uri = file:///home/joakim/code/jetty/distros/bases/prop-base
 jetty.home = /home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home
 jetty.home.uri = file:///home/joakim/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home
 jetty.sslContext.keyStorePassword = OBF:1vny1zlo1x8e1vnw1vn61x8g1zlu1vn4
 jetty.sslContext.keyStorePath = etc/test-keystore.p12
 jetty.sslContext.keyStoreType = PKCS12
 jetty.sslContext.trustStorePath = etc/test-keystore.p12
 jetty.sslContext.trustStoreType = PKCS12
 jetty.webapp.addServerClasses = org.eclipse.jetty.logging.,org.slf4j.,${jetty.base.uri}/lib/bouncycastle/
 runtime.feature.alpn = true
 slf4j.version = 2.0.0-alpha1

Jetty Server Classpath:
-----------------------
Version Information on 12 entries in the classpath.
Note: order presented here is how they would appear on the classpath.
      changes to the --module=name command line options will be reflected here.
 0:                    (dir) | ${jetty.base}/resources
 1:             2.0.0-alpha1 | ${jetty.home}/lib/logging/slf4j-api-2.0.0-alpha1.jar
 2:          10.0.2-SNAPSHOT | ${jetty.home}/lib/logging/jetty-slf4j-impl-10.0.2-SNAPSHOT.jar
 3:                    4.0.6 | ${jetty.home}/lib/jetty-servlet-api-4.0.6.jar
 4:          10.0.2-SNAPSHOT | ${jetty.home}/lib/jetty-http-10.0.2-SNAPSHOT.jar
 5:          10.0.2-SNAPSHOT | ${jetty.home}/lib/jetty-server-10.0.2-SNAPSHOT.jar
 6:          10.0.2-SNAPSHOT | ${jetty.home}/lib/jetty-xml-10.0.2-SNAPSHOT.jar
 7:          10.0.2-SNAPSHOT | ${jetty.home}/lib/jetty-util-10.0.2-SNAPSHOT.jar
 8:          10.0.2-SNAPSHOT | ${jetty.home}/lib/jetty-io-10.0.2-SNAPSHOT.jar
 9:          10.0.2-SNAPSHOT | ${jetty.home}/lib/jetty-keystore-10.0.2-SNAPSHOT.jar
10:                 1.62.0.0 | ${jetty.base}/lib/bouncycastle/bcpkix-jdk15on-1.62.jar
11:                   1.62.0 | ${jetty.base}/lib/bouncycastle/bcprov-jdk15on-1.62.jar

Jetty Active XMLs:
------------------
 ${jetty.home}/etc/jetty-bytebufferpool.xml
 ${jetty.home}/etc/jetty-threadpool.xml
 ${jetty.home}/etc/jetty.xml
 ${jetty.home}/etc/jetty-ssl.xml
 ${jetty.home}/etc/jetty-ssl-context.xml
 ${jetty.home}/etc/jetty-https.xml
 ${jetty.home}/etc/jetty-test-keystore.xml

```

**Comment by joakime** [comment]
I don't like seeing bouncycastle present in the server classpath from that simple `--add-modules=ssl,https,test-keystore`

**Comment by joakime** [comment]
Looks like the `test-keystore.mod` was only created for the demo to use.
Perhaps it should be renamed to `demo-keystore.mod` ?

**Comment by sbordet** [comment]
@joakime no, the `test-keystore` is used in the docs and for whoever wants to try Jetty without a demo or getting a real KeyStore.

**Comment by joakime** [comment]
The error is from the attempt to load the TrustStore in ...

https://github.com/eclipse/jetty.project/blob/d9eefc92313ea6c7bfc4505600da0fa5ad5b3b07/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java#L1113-L1125

The resource `jetty.sslContext.trustStorePath = etc/test-keystore.p12`, doesn't exist either.

**Comment by joakime** [comment]
If you set both to `bogus` it doesn't complain.

```
$ java -jar ~/code/jetty/jetty.project-10.0.x/jetty-home/target/jetty-home/start.jar jetty.sslContext.keyStorePath=bogus jetty.sslContext.trustStorePath=bogus 
2021-03-01 12:14:38.635:WARN :oejk.KeystoreGenerator:main: Generating Test Keystore: DO NOT USE IN PRODUCTION!
2021-03-01 12:14:39.098:INFO :oejs.Server:main: jetty-10.0.2-SNAPSHOT; built: 2021-03-01T17:28:03.033Z; git: d9eefc92313ea6c7bfc4505600da0fa5ad5b3b07; jvm 11.0.10+9
2021-03-01 12:14:39.205:INFO :oejus.SslContextFactory:main: x509=X509@584f54e6(jetty-test-keystore,h=[localhost],w=[]) for Server@f80945f[provider=null,keyStore=file:///home/joakim/code/jetty/distros/bases/prop-base/bogus,trustStore=file:///home/joakim/code/jetty/distros/bases/prop-base/bogus]
2021-03-01 12:14:39.285:INFO :oejs.AbstractConnector:main: Started ServerConnector@4451366b{SSL, (ssl, http/1.1)}{0.0.0.0:8443}
2021-03-01 12:14:39.300:INFO :oejs.Server:main: Started Server@6155d082{STARTING}[10.0.2-SNAPSHOT,sto=5000] @1464ms
```

**Comment by joakime** [comment]
I wonder _why we don't throw an exception_ if a keystore resource is configured in the SslContextFactory that doesn't exist?

**Comment by gregw** [comment]
@sbordet what is the status of this? Can we close?

**Comment by sbordet** [comment]
The problem is present in 9.4.x too, and it's due to the presence of an `[ini]` section.

**Comment by gregw** [comment]
@sbordet can this be bumped 9.4.45?

**Comment by joakime** [comment]
@gregw bumped

**Comment by joakime** [comment]
Fix in ec3d7dbe92532cfdbb719cb7b736ad115aea40e5

**Comment by joakime** [comment]
@sbordet discovered that `key+=value` is broken, likely as a result of this change.
Reopening.
